### ISSUE TITLE
_create_composite_credentials creates a non async requests

### ISSUE THREAD

**Issue by ArcLightSlavik** [googleapis/python-api-core#151]
In `grpc_helpers.py`:

```
def _create_composite_credentials(
        credentials=None,
        credentials_file=None,
        default_scopes=None,
        scopes=None,
        ssl_credentials=None,
        quota_project_id=None,
        default_host=None):
    # removed stuff before this
    request = google.auth.transport.requests.Request()

    # Create the metadata plugin for inserting the authorization header.

    # TODO: remove this if/else once google-auth >= 1.25.0 is required
    if _GOOGLE_AUTH_HAS_DEFAULT_SCOPES_AND_DEFAULT_HOST:
        metadata_plugin = google.auth.transport.grpc.AuthMetadataPlugin(
            credentials, request, default_host=default_host,
        )
    else:
        metadata_plugin = google.auth.transport.grpc.AuthMetadataPlugin(
            credentials, request
        )

    # Create a set of grpc.CallCredentials using the metadata plugin.
    google_auth_credentials = grpc.metadata_call_credentials(metadata_plugin)

    if ssl_credentials is None:
        ssl_credentials = grpc.ssl_channel_credentials()

    # Combine the ssl credentials and the authorization credentials.
    return grpc.composite_channel_credentials(
        ssl_credentials, google_auth_credentials
    )
```

This method gets called from `grpc_helpers_async.py`:
```
def create_channel(
        target,
        credentials=None,
        scopes=None,
        ssl_credentials=None,
        credentials_file=None,
        quota_project_id=None,
        default_scopes=None,
        default_host=None,
        **kwargs):

    composite_credentials = grpc_helpers._create_composite_credentials(
        credentials=credentials,
        credentials_file=credentials_file,
        scopes=scopes,
        default_scopes=default_scopes,
        ssl_credentials=ssl_credentials,
        quota_project_id=quota_project_id,
        default_host=default_host
    )

    return aio.secure_channel(target, composite_credentials, **kwargs)
```

But the `request = google.auth.transport.requests.Request()` is based on `requests` which isn't async, google.auth.transport does have a aiohttp transport <https://github.com/googleapis/google-auth-library-python/blob/master/google/auth/transport/_aiohttp_requests.py>

**Comment by busunkim96** [comment]
@ArcLightSlavik Thanks for the report.

Is the request causing issues for you when you use an AsyncIO channel? 

If I'm reading the gRPC docs correctly, it looks like [`grpc.AuthMetadataPlugin.__call__`](https://github.com/googleapis/google-auth-library-python/blob/e87b1eca82ffc5ca9d2704b04656aa4e869547bf/google/auth/transport/grpc.py#L96) is invoked asynchronously in a separate thread (In both the sync/async implementations)

https://grpc.github.io/grpc/python/grpc.html#grpc.AuthMetadataPlugin

**Comment by ArcLightSlavik** [comment]
Don't think it causes any issues directly.

We did have issues with test freezing due to auth but that was fixed by upgrading to grcpio 1.35.

**Comment by busunkim96** [comment]
Hi,

I'm going to close out the issue as I believe the original question has been addressed. Please open a new issue if you are still experiencing problems.

Thank you!
